middleware/Authenticate.js

```diff
@@ -0,0 +1,37 @@
+const jwt = require("jsonwebtoken");
+const User = require("../model/userSchema");
+const cookieParser = require('cookie-parser');
+const express = require('express');
+const app = express();
+
+app.use(cookieParser());
+
+
+//this function mainly aims to restores the session of already signed-in user, 
+// uses a simple technique to find a user's details from token, if no user is found then accordingly 
+// throw an error
+const Authenticate = async (req, res, next) => {
+    try {
+        console.log(req.cookies);
+        const token = req.cookies.jwtoken;
+        
+        const verifyToken = jwt.verify(token, process.env.SECRET_KEY);
+
+        const rootUser = await User.findOne({ _id: verifyToken._id, "tokens.token": token });
+
+        if (!rootUser) { throw new Error('User not Found') }
+        
+        req.token = token;
+        req.rootUser = rootUser;
+        req.userID = rootUser._id;
+
+        next();
+        
+    } catch (err) {
+        res.status(401).send('Unauthorized:No token provided');
```

Comment:
response handling should be done seperatley rather than passing arbitary strings and status codes everywhere in your files. will save time for more business critical things to work on.
@Mitali-laroia @zeel-codder

Reply:
Got your point, but how should we ideally handle response handling especially in those cases where we need to check the errors, any industry standards, or demo code would really help us keeping this in mind and implementing the same.

Reply:
Sure, currently we are merging this PR, for the very next issue we'll be working on we'll refactor the code